manager/apis/app/v1alpha1/m4dstorageaccount_types.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// M4DStorageAccountSpec defines the desired state of M4DStorageAccount
+type M4DStorageAccountSpec struct {
+	// +required
+	// Secret including credentials for COS bucket
+	SecretRef string `json:"secretRef"`
+	// +required
+	// Endpoint
+	Endpoint string `json:"endpoint"`
+	// +required
+	// +kubebuilder:validation:MinItems=1
+	// Regions
+	Regions []string `json:"regions"`
```

Comment:
FYI I think that endpoints and regions are usually coupled but ok for now, we can change it later as needed.

Reply:
OK, so one set of credentials for multiple <endpoint,region> pairs?

Reply:
I think so but I'm not sure if that's true on all/most major cloud providers so we can keep it for later PR after we verify.